# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: D:\Profiles\akm022\NetBeansProjects\main\core
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: src/main/java/hudson/model/AbstractProject.java
--- src/main/java/hudson/model/AbstractProject.java Base (BASE)
+++ src/main/java/hudson/model/AbstractProject.java Locally Modified (Based On LOCAL)
@@ -1178,7 +1178,7 @@
         R lb = getLastBuild();
         if (lb==null) {
             listener.getLogger().println(Messages.AbstractProject_NoBuilds());
-            return BUILD_NOW;
+            return isInQueue() ? NO_CHANGES : BUILD_NOW;
         }
 
         if (pollingBaseline==null) {
@@ -1213,8 +1213,13 @@
                     listener.getLogger().println( ws==null
                         ? Messages.AbstractProject_WorkspaceOffline()
                         : Messages.AbstractProject_NoWorkspace());
+                    if (isInQueue()) {
+                        listener.getLogger().println(Messages.AbstractProject_AwaitingBuildForWorkspace());
+                        return NO_CHANGES;
+                    } else {
                     listener.getLogger().println(Messages.AbstractProject_NewBuildForWorkspace());
                     return BUILD_NOW;
+                    }
                 } else {
                     WorkspaceList l = lb.getBuiltOn().toComputer().getWorkspaceList();
                     // if doing non-concurrent build, acquire a workspace in a way that causes builds to block for this workspace.
Index: src/main/resources/hudson/model/Messages.properties
--- src/main/resources/hudson/model/Messages.properties Base (BASE)
+++ src/main/resources/hudson/model/Messages.properties Locally Modified (Based On LOCAL)
@@ -27,6 +27,7 @@
 
 AbstractItem.NoSuchJobExists=No such job ''{0}'' exists. Perhaps you meant ''{1}''?
 AbstractProject.NewBuildForWorkspace=Scheduling a new build to get a workspace.
+AbstractProject.AwaitingBuildForWorkspace=Awaiting build to get a workspace.
 AbstractProject.Pronoun=Project
 AbstractProject.Aborted=Aborted
 AbstractProject.BuildInProgress=Build #{0} is already in progress{1}
